conformance_tests/sysman/test_sysman_device/src/test_sysman_device_helper.cpp

```diff
@@ -74,22 +106,40 @@ int main(int argc, char **argv) {
   auto sysman_devices = lzt::get_zes_devices();
   EXPECT_FALSE(sysman_devices.empty());
 
-  std::vector<zes_uuid_t> sysman_device_uuids;
-  get_sysman_devices_uuids(sysman_devices, sysman_device_uuids);
+  if (strcmp(device_hierarchy, "FLAT") != 0) { // composite or combined mode
+    std::vector<UUID> sysman_device_uuids;
+    std::vector<UUID> ze_root_uuids;
 
-  std::vector<ze_device_uuid_t> ze_root_uuids;
-  get_ze_root_uuids(ze_devices, ze_root_uuids, device_hierarchy);
+    for (const auto &sysman_device : sysman_devices) {
+      get_sysman_device_uuid(sysman_device, sysman_device_uuids);
+    }
+
+    for (const auto &ze_device : ze_devices) {
+      get_ze_root_uuid(ze_device, ze_root_uuids, device_hierarchy);
+    }
 
-  for (const auto &ze_root_uuid : ze_root_uuids) {
-    bool ze_and_sysman_uuid_equal = false;
-    for (const auto &sysman_device_uuid : sysman_device_uuids) {
-      if (memcmp(ze_root_uuid.id, sysman_device_uuid.id,
-                 ZE_MAX_DEVICE_UUID_SIZE) == false) {
-        ze_and_sysman_uuid_equal = true;
-        break;
+    compare_core_and_sysman_uuid(ze_root_uuids, sysman_device_uuids);
+  } else { // flat mode
+    std::vector<UUID> sysman_device_uuids;
+    std::vector<UUID> ze_device_uuids;
+
+    for (const auto &sysman_device : sysman_devices) {
+      auto device_properties = lzt::get_sysman_device_properties(sysman_device);
+      uint32_t sub_devices_count = device_properties.numSubdevices;
+      if (sub_devices_count > 0) {
+        get_sysman_sub_devices_uuids(sysman_device, sysman_device_uuids);
+      } else {
+        // if subdevice doesn't exist for a device, then root device UUID
+        // is retrieved to match with core UUID's retrieved from flat mode
+        get_sysman_device_uuid(sysman_device, sysman_device_uuids);
       }
     }
-    EXPECT_TRUE(ze_and_sysman_uuid_equal);
+
+    for (const auto &ze_device : ze_devices) {
+      ze_device_uuids.push_back(get_ze_device_uuid(ze_device));
+    }
+
+    compare_core_and_sysman_uuid(ze_device_uuids, sysman_device_uuids);
```

Comment:
Can we call these outside if branching ?

Reply:
Kept inside for the same reason mentioned above

Reply:
Updated as per discussion